src/controller.rs

```diff
@@ -114,27 +120,69 @@ async fn reconcile(app_instance: Arc<AppInstance>, ctx: Arc<Context>) -> Result<
         serde_json::from_value(config.metadata.get(PACK_KEY).unwrap().clone())
             .map_err(Error::DecodeKubecfgPackageMetadata)?;
 
-    launch_job(&app_instance, &kubecfg_pack_metadata, Arc::clone(&ctx)).await?;
+    let configmap_name = create_configmap_overlay(&app_instance, Arc::clone(&ctx)).await?;
+    launch_job(
+        &app_instance,
+        &kubecfg_pack_metadata,
+        configmap_name,
+        Arc::clone(&ctx),
+    )
+    .await?;
 
     Ok(Action::await_change())
 }
 
+fn calculate_hash<T: Hash>(t: &T) -> u64 {
+    let mut s = DefaultHasher::new();
+    t.hash(&mut s);
+    s.finish()
+}
+
+async fn create_configmap_overlay(app_instance: &AppInstance, ctx: Arc<Context>) -> Result<String> {
+    let overlay_obj = serde_json::to_string(&app_instance).map_err(Error::RenderOverlay)?;
+    let overlay_hash = calculate_hash(&overlay_obj.as_bytes());
+
+    let mut configmap_data: BTreeMap<String, String> = BTreeMap::new();
+    configmap_data.insert("overlay.json".to_string(), overlay_obj);
+
+    let ns = app_instance.clone().namespace().unwrap();
+    let configmap_name = format!("kubit-overlay-{overlay_hash}");
+
+    let configmaps: Api<ConfigMap> = Api::namespaced(ctx.client.clone(), &ns);
+    let cm = ConfigMap {
+        metadata: ObjectMeta {
+            name: Some(configmap_name),
+            namespace: app_instance.namespace().clone(),
+            ..Default::default()
+        },
+        data: Some(configmap_data),
+        ..Default::default()
+    };
+
+    let pp = PostParams::default();
+
+    match configmaps.create(&pp, &cm).await {
+        Ok(o) => info!("Created ConfigMap: {}", o.name_any()),
+        Err(kube::Error::Api(ae)) => match ae.code {
+            409 => info!("configmap already exist, doing nothing"),
+            _ => return Err(kube::Error::Api(ae).into()),
+        },
+        Err(e) => panic!("API error: {}", e),
+    }
+
+    Ok(cm.metadata.name.unwrap())
+}
+
 async fn launch_job(
     app_instance: &AppInstance,
     kubecfg_pack_metadata: &KubecfgPackageMetadata,
+    configmap_name: String,
```

Comment:
I was considering passing in the entire `ConfigMap` struct here, but it seems relatively pointless as it isn't used at all. All we really care about is the name of it so that it can be used as a `volume` in the `Job`.

Happy to alter this though as there is likely a better way to handle it.